tests/commands.rs

```diff
@@ -80,6 +83,91 @@ fn fuelup_toolchain_install_nightly() -> Result<()> {
     Ok(())
 }
 
+#[test]
+fn fuelup_toolchain_install_nightly_date() -> Result<()> {
+    testcfg::setup(FuelupState::Empty, &|cfg| {
+        cfg.fuelup(&["toolchain", "install", "nightly-2022-08-31"]);
+
+        for entry in cfg.toolchains_dir().read_dir().expect("Could not read dir") {
+            let toolchain_dir = entry.unwrap();
+            let expected_toolchain_name =
+                "nightly-2022-08-31-".to_owned() + &TargetTriple::from_host().unwrap().to_string();
+            assert_eq!(
+                expected_toolchain_name,
+                toolchain_dir.file_name().to_str().unwrap()
+            );
+            assert!(toolchain_dir.file_type().unwrap().is_dir());
+
+            expect_files_exist(&toolchain_dir.path().join("bin"), ALL_BINS);
+        }
+    })?;
+
+    Ok(())
```

Comment:
Why do these tests return an empty tuple (why do they return a result at all)?

Reply:
Each test does some setup prior to executing the actual tests, and within the setup we execute `setup_toolchain` which returns a Result and may fail, because it is trying to create tmp files/directories for tests. So I had to use the `?` notation for the setup of each test (which requires that the test fn returns a `Result`).